crates/dbsp/src/operator/input.rs

```diff
@@ -683,7 +693,7 @@ where
                 partition_size += 1;
             }
 
-            let worker = (next_worker + i) % num_partitions;
+            let worker = (next_worker + i) % num_partitions + self.input_handle.workers().start;
```

Comment:
The compiler will likely take care of this, but I would still pre-compute `self.input_handle.workers().start` outside the loop.

Reply:
Done.